### ISSUE TITLE
Add a quick watch popup for easy debugging of expressions

### ISSUE THREAD

**Issue by filipdutescu** [helix-editor/helix#5955]
# Description

A common feature in modern IDEs, which is very useful for quick evaluations of expression, without the need to add (and most likely edit and delete) a watch is quick watch (or evaluate). It shows the values of expressions using the current scope as a context for variables, functions etc. 

# Proposal

Add a quick watch popup, which could start empty or from the current code selection, behaving similar to a picker, with expandable values for the result of the expression. Expandable values should follow the implementation of locals from #5954.

# UI mockups

![image](https://user-images.githubusercontent.com/46979062/218332381-4ce5120d-039a-4b5b-b564-905eb6a08096.png)

**Comment by pascalkuthe** [comment]
I belive https://github.com/helix-editor/helix/pull/5768 also contains a similar overlay tree view in addition to the side tree view so again basing your work on that pr will help

**Comment by filipdutescu** [comment]
Yep, will do so, this change will probably depend on #5954, since after that one this one will probably be relatively trivial to achieve.
